src/hal/src/buffer.rs

```diff
@@ -15,6 +15,27 @@ pub enum CreationError {
     Other,
 }
 
+impl fmt::Display for CreationError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        let description = self.description();
+        match *self {
+            CreationError::Usage(usage) => write!(f, "{}: {:?}", description, usage),
+            _ => write!(f, "{}", description)
+        }
+    }
+}
+
+impl Error for CreationError {
+    fn description(&self) -> &str {
+        match *self {
+            CreationError::Usage(_) =>
+                "Required `Usage` is not supported",
+            CreationError::Other =>
+                "Some other problem",
```

Comment:
Lol, I don't think it's appropriate descrtiption here.

Reply:
I was thinking exacly same. But I just copied it from comment for variant.